src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.cs

```diff
@@ -75,6 +80,408 @@ public static int IndexOfQuoteOrAnyControlOrBackSlash(this ReadOnlySpan<byte> sp
+                                            Avx2.Or(
+                                                Avx2.CompareGreaterThan(
+                                                    valuesLessThan,
+                                                    Avx2.Subtract(search, Vector256.Create((byte)0x80)).AsSByte()).AsByte(),
+                                                Avx2.Or(
+                                                    Avx2.CompareEqual(values0, search),
+                                                    Avx2.CompareEqual(values1, search))
+
+                                                )
+                                            );
+                            // Note that MoveMask has converted the equal vector elements into a set of bit flags,
+                            // So the bit position in 'matches' corresponds to the element offset.
+                            if (matches == 0)
+                            {
+                                // None matched
+                                offset += (nuint)Vector256<byte>.Count;
+                                continue;
+                            }
+
+                            goto IntrinsicsMatch;
+                        }
+
+                        {
+                            // Move to Vector length from end for final compare
+                            Vector256<byte> search = LoadVector256(ref searchSpace, lengthToExamine);
+                            offset = lengthToExamine;
+                            // Same as method as above
+                            matches = Avx2.MoveMask(
+                                            Avx2.Or(
+                                                Avx2.CompareGreaterThan(
+                                                    valuesLessThan,
+                                                    Avx2.Subtract(search, Vector256.Create((byte)0x80)).AsSByte()).AsByte(),
+                                                Avx2.Or(
+                                                    Avx2.CompareEqual(values0, search),
+                                                    Avx2.CompareEqual(values1, search))
+
+                                                )
+                                            );
+                        }
+
+                        if (matches == 0)
+                        {
+                            // None matched
+                            goto NotFound;
+                        }
+
+                        goto IntrinsicsMatch;
+                    }
+                }
+
+                // Initial size check was done on method entry.
+                Debug.Assert(length >= Vector128<byte>.Count);
+                {
+                    Vector128<byte> values0 = Vector128.Create(value0);
+                    Vector128<byte> values1 = Vector128.Create(value1);
+                    Vector128<sbyte> valuesLessThan = Vector128.Create((sbyte)(uLessThan - 0x80));
```

Comment:
AVX2 assumed:
This is essentially recreating the vectors which are already there (L221-L223), so we could reuse them to eliminate some unnecessry broadcasts. See [sharplap](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8ovLrl5hcAbhr0mrTj36CRGMRKm4WADQAcSOdXkBmJqQYBhBgG8aDR03PiMDALIAKYAE8MMBmByAEo7BycI3gAzBk8AZVwYUmFcOI4AB3TofwATYPCIx3tqQtKGaNiAQQA3BGShVIysqFz8krKI4o6OgDUYMAxoUgBWJAAeX38APgZq/pGkBgBeBj6BodGWK1hsf28Q426ygqOnCs8aupZ3CDn3PABrTzmwBdCllbo208Kun7LiAB2BgoUiHf6OAC+Jw60PaRxhZTWgyg5FI+gmfhgMxeaP0y1W/RReK2Oz2QWC4IR8O6QIYCSS11uMHuuCeuPRlMRUO5DF5dLg5CpDDhEROxBcPA85G8WMCITCNMiMXiiXqjUy2RgeV5fyOyI240m2Nm81GBJyMCi2A4ABsMMKOrzlVVauq0pqWtrvqc9f8XgsCQaoAtSTBdjB9lyladnWVzpdkjc7o9nmakO9Pj6IX6IY46aDHadRcW444y4T1qj0ZjpqawHiCYmUh7mrkKxEAPz10MAcRgGAAMhAAO4wKCebP/ECV4nosMRqNF0oVukMpPM1ns/p46Ow3n84GC4VwyFAA) for an outline of the idea.
(I think / hope the extra `vmovaps ymm1, ymm0` in the sharplab-asm isn't generated by current JIT.)

If AVX2 isn't available the JIT will remove the dead artifacts, and codegen is good.

---

I used this "trick" on several places (mostly proprietary work), and has some positive effects on smaller inputs, where the overhead of broadcasting can show up.

You could use this in your other PRs too? (Sorry for not mentioning this on that PRs, because I didn't look too diligent...)

Reply:
It takes an either/or path based on length; so the broadcasts are only called once (Avx2 doesn't then move to Sse2); so would only reduce code gen size?

Reply:
> Avx2 doesn't then move to Sse2

Ah, I missed that. 
Generally I like the pattern AVX -> SSE -> scalar, as the SSE-path could still "eat" some data as opposed to AVX | SSE -> scalar. Maybe it's worth to give this a try (although I assume you have already explored that).

> so would only reduce code gen size?

If there are no spills (I expect there won't be any) then yes.